src/scenes/Languages/Edit/EditLanguageForm.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { UpdateLanguageInput } from '../../../api';
+import {
+  DialogForm,
+  DialogFormProps,
+} from '../../../components/Dialog/DialogForm';
+import { SubmitError, TextField } from '../../../components/form';
+import { LanguageQuery } from '../Overview/LanguageOverview.generated';
+
+export type EditLanguageFormProps = DialogFormProps<UpdateLanguageInput> & {
+  languageFormValues: UpdateLanguageInput;
+  language?: LanguageQuery['language'];
+};
+
+export const EditLanguageForm = (props: EditLanguageFormProps) => {
+  const { language, languageFormValues } = props;
+
+  return (
+    <DialogForm
+      DialogProps={{
+        fullWidth: true,
+        maxWidth: 'xs',
+      }}
+      SubmitProps={{ disabled: !language }}
+      initialValues={languageFormValues}
+      {...props}
+      title="Edit Language"
+    >
+      <SubmitError />
+      <TextField
+        name="language.name"
+        label="Name"
+        placeholder="Enter language"
+        autoFocus
+        required
+        disabled={!language?.name.canEdit}
+      />
+      <TextField
+        name="language.displayName"
+        label="Display Name"
+        placeholder="Enter Display Name"
+        required
+        disabled={!language?.displayName.canEdit}
+      />
+      <TextField
+        name="language.beginFiscalYear"
+        label="Fiscal Year"
+        placeholder="Enter Fiscal Year Begun"
+        type="number"
+        disabled={!language?.beginFiscalYear?.canEdit}
+      />
+      <TextField
+        name="language.ethnologueName"
+        label="Ethnologue Name"
+        placeholder="Enter Ethnologue Name"
+        disabled={!language?.ethnologueName?.canEdit}
+      />
+      <TextField
```

Comment:
If you rebase this branch on v3 you can use the new `NumberField` here

Reply:
Changed corresponding fields to use this